### ISSUE TITLE
bug: `detect_mathlib` doesn't find mathlib dependency

### ISSUE THREAD

**Issue by Seasawher** [leanprover/lean-action#85]
I use this action on my repository [lean-by-example](https://github.com/lean-ja/lean-by-example). Today this action fails to detect mathlib dependency...

![image](https://github.com/user-attachments/assets/08b579f0-bb9b-4165-9426-cab80e603498)

full workflow log: https://github.com/lean-ja/lean-by-example/actions/runs/10436340213/job/28901291019

**Comment by Seasawher** [comment]
why this workflow fail to detect mathlib...?

logs of CI workflow: https://github.com/lean-ja/lean-by-example/actions/workflows/ci.yml

By looking at the execution time of the workflow, it can be seen that it is suddenly failing when previously it was fine.

**Comment by Seasawher** [comment]
I understand why this issue ocurrs. `detect_mathlib` cannot treat new lakefile syntax.

**Comment by austinletson** [comment]
> I now understand why this issue ocurrs. `detect_mathlib` cannot treat new lakefile syntax.
> 
> this issue should be resolved by fixing #82.

@Seasawher, I believe you are correct about the cause and the right fix here.

I am working on a fix for #82. Note that in the meantime if this is causing major issues, you can use 
```yml
        uses: leanprover/lean-action@v1.0.0
        with:
          build-args: "--log-level=error"
          use-mathlib-cache: "true"
```

**Comment by austinletson** [comment]
Fixed by #88
